### ISSUE TITLE
Settings general.default_host and general.default_ttl settings broken

### ISSUE THREAD

**Issue by cthorner** [fastly/fastly-ruby#63]
I'm getting an error when trying to set the settings for an existing service:

```
settings = fastly.update_settings( service_id: service.id, version: latest_version.number, settings: {"general.default_host"=>"www.example.com", "general.default_ttl" => 500})
NoMethodError: undefined method `service_id' for #<Hash:0x0000000dfaecb0>
from /home/vagrant/.rvm/gems/ruby-2.0.0-p353/gems/fastly-1.1.5/lib/fastly/settings.rb:31:in `put_path'
```

I may be using the update_settings call wrong, but from the code it looks like those variables should be available.

**Comment by set5think** [comment]
This seems like a bug in terms of `update_settings` suggesting that `opts` should be a Hash, as the Setting object's `put_path` method is looking for the argument to be an object, not a hash (https://github.com/fastly/fastly-ruby/blob/7d2e13dc5f894499fdbcac674444338e860894f8/lib/fastly/settings.rb#L31).

Technically, what you could do is create a `Settings` instance and pass it to the `update_settings` method (until this gets fixed, and likely an interface change so you don't have to do this hack):

```
settings = fastly.get_settings('service_id', 'version_number') # this should create an instance of `Fastly::Settings`

settings.version = latest_version.number # up to you whether you need to do this or not
settings.settings = {"general.default_host"=>"www.example.com", "general.default_ttl" => 500}

settings.update_settings(settings)
```

Super gross, but it seems like it should work.

**Comment by cthorner** [comment]
Yes, I noticed that too. I tried using OpenStruct to get around this, but got a different error. Your method worked, but the last statement should be

```
fastly.update_settings(settings)
```

Thanks!

**Comment by set5think** [comment]
Cool, glad it worked.

**Comment by Integralist** [comment]
Closing this issue now we've published v4.0.0 https://rubygems.org/gems/fastly
